### ISSUE TITLE
Make summaries visible by default

### ISSUE THREAD

**Issue by willguv** [localgovdrupal/localgov#243]
Summary fields across LocalGov Drupal do not currently behave in a consistent way. 

While working at Cumbria, I asked our developer (@danchamp) to investigate current behaviour and propose fixes where required - see below for conversation

**Comment by danchamp** [comment]
@willguv I've started looking at this. Making the summary fields open by default is straightforward, but there was also mention of improving the help text on back-end forms (specifically for summaries, and there's a wider need for editor guidance).

Before I start down that road it would be useful to get input on what will be useful, and assumptions we can make. For example, my first effort for the directory channel summary/body text looks like this:

<img width="765" alt="image" src="https://user-images.githubusercontent.com/261421/120338990-a28aae00-c2ec-11eb-9b11-16226d3d176f.png">

Leaving aside the presentation for now (I'd like to have separate help text for the each of the summary and body fields, but out of the box we are very limited in what we can do):

`The channel Summary is presented as a lede below the channel title; Body is presented at the top of the channel content.`

2 questions:
 
1. Is this the sort of information that is going to help content editors?
2. Does it matter that we're assuming the content will be rendered consistently with the base theme? In this case it's unlikely to be different, but for something like news the use of the summary may vary from council to council.

If we can agree some principles I'll go away and create and collate help text for all content types in a format that can be reviewed before committing to code.

**Comment by willguv** [comment]
I'm wondering if we can ask some of our content designers to write this text? @benhillsjones do you have time?

Before we start on summary and body helper text, I'm keen to find out whether the use of them is consistent across the site. So there might be some discovery, thinking and fixing before we can properly dive in to writing helper text.

If there's work to see, @danchamp shall we look at this at the end of tomorrow's Cumbria meet?

**Comment by danchamp** [comment]
> I'm wondering if we can ask some of our content designers to write this text? @benhillsjones do you have time?

Definitely, that would be much better.

> So there might be some discovery, thinking and fixing before we can properly dive in to writing helper text.

Agreed.

> If there's work to see, @danchamp shall we look at this at the end of tomorrow's Cumbria meet?

Great, thanks.

**Comment by danchamp** [comment]
@willguv A collation of the current state of summary fields across content types, and suggested actions for rationalisation:

| Content type             | Visible? | Required? | Usage                                                        | Suggested action                                           |
|--------------------------|----------|-----------|--------------------------------------------------------------|------------------------------------------------------------|
| Directory channel        | No       | No        | Lede on page header block                                    | Make visible ([PR](https://github.com/localgovdrupal/localgov_directories/pull/107) / [PR 1.x](https://github.com/localgovdrupal/localgov_directories/pull/108))                                          |
| Directory page           | No       | No        | Lede on page header block Directory listing in channel       | Make visible ([PR](https://github.com/localgovdrupal/localgov_directories/pull/107) / [PR 1.x](https://github.com/localgovdrupal/localgov_directories/pull/108))                                          |
| Directory venue          | No       | No        | Lede on page header block Directory listing in channel       | Make visible ([PR](https://github.com/localgovdrupal/localgov_directories/pull/107) / [PR 1.x](https://github.com/localgovdrupal/localgov_directories/pull/108))                                          |
| Event                    | No       | No        | Lede on page header block                                    | Make visible                                               |
| Guide overview           | No       | No        | Lede on page header block                                    | Make visible                                               |
| Guide page               | No       | No        | Lede on page header block                                    | Make visible, remove redundant description field (see [issue #33](https://github.com/localgovdrupal/localgov_guides/issues/33)) |
| News article             | Yes      | Yes       | Newsroom featured & list                                     | None                                                       |
| Service landing page     | Yes      | Yes       | Lede on page header block Services view block                | None                                                       |
| Service sub-landing page | Yes      | Yes       | Lede on page header block                                    | None                                                       |
| Service page             | Yes      | Yes       | Lede on page header block                                    | None                                                       |
| Service status           | Yes      | Yes       | Lede on page header block Service status view block and page | None                                                       |
| Step-by-step overview    | No       | No        | Lede on page header block                                    | Make visible                                               |
| Step-by-step page        | Yes      | No        | Used in summaries block                                      | None                                                       |
| Subsite overview         | Yes      | No        | Unused on subsites                                           | None                                                       |
| Subsite page             | Yes      | No        | Unused on subsites                                           | None                                                       |

**Comment by andybroomfield** [comment]
Still live issue

**Comment by andybroomfield** [comment]
Ref https://github.com/localgovdrupal/localgov/issues/487

**Comment by willguv** [comment]
Thanks for flagging this @andybroomfield - have moved into our new BAU backlog

@keelanfh have you done any of this work?

@benhillsjones could we discuss at a content group soon?

Cheers all

**Comment by keelanfh** [comment]
I don't think I've really looked at this. It has come up as a point of confusion, especially localgovdrupal/localgov_guides#33 mentioned above which comes up repeatedly.

**Comment by willguv** [comment]
@keelanfh would these names suit your team better?

Landing Page | service_landing_page
Sub Landing Page | service_sub_landing_page
Page | service_page

I believe the content group has mentioned content format names too - @benhillsjones do you have details?

**Comment by andybroomfield** [comment]
@willguv I think @keelanfh is referencing https://github.com/localgovdrupal/localgov_guides/issues/33

**Comment by keelanfh** [comment]
Ah thanks @andybroomfield, yes that's right - I got a bit confused by @willguv's response! Blame it on it being Friday afternoon!

**Comment by willguv** [comment]
Oh I see, sorry for confusion - I'll make a new card for content format names if needed

@andybroomfield does your PR remove the guide description in favour of the summary field?

**Comment by keelanfh** [comment]
From Essex point of view, I don't think we've had any issues raised with content format names.

**Comment by andybroomfield** [comment]
The PR on guides would have fixed the lede, but I gather the descion here is to make sure all summary fields are the same which is what BHCC would support, so new PR (and possible migration needed).

**Comment by willguv** [comment]
Notes from content group

Ben: Summary fields are often hard to miss so they're closed. They should always be open

Lizzie: Easy to miss on events content type too

There's a general principle about making summaries mandatory

Check for oddities eg on step by step

**Comment by willguv** [comment]
@stephen-cox there's a few changes here, but let's start with the easy one!

Can you make summaries visible by default on the following content formats please:

- event
- guide overview
- step by step overview

I'll create a few separate related issues. Thanks again

**Comment by keelanfh** [comment]
Is this likely to have an update hook to change existing installations? Or just for new ones? 

(Just figuring out if it's something I need in my backlog 😆)

**Comment by andybroomfield** [comment]
I belive it will require an update hook, I don't think guides / step by step should have content in the summary, but its worth checking and deciding how we are going to merge them if we push an update to existing councils. 
Since the summary is (or should be?) used for the lede in the page header, on the summary of page navigation cards and if the source for that changes then they will need to updated.

I think the issue is if someone has different content in both fields, thats going to be a pain point to manage and the update process should abort? /cc @stephen-cox @ekes

**Comment by stephen-cox** [comment]
I think making the summaries visible will just be for new installs. Current installs may have customised things in ways we can't anticipate and so while it's a simple change it could be detrimental to editor experience if an unused summary suddenly appears.

If we start merging and removing fields this might get very tricky to apply to current installs, but it's certainly something that can be looked at if there's enough interest, but we would have to check things are in a known state before migrating stuff. The issue then is we'll need support both the old and new configurations; which might just be a theme thing, but could extend beyond that.

**Comment by willguv** [comment]
Hi @stephen-cox could we provide documentation for existing installs to make these changes themselves. Are they straightforward to do?

Thanks

**Comment by stephen-cox** [comment]
There are now issues for making the summaries open by default:

- https://github.com/localgovdrupal/localgov_guides/issues/110
- https://github.com/localgovdrupal/localgov_step_by_step/issues/75
- https://github.com/localgovdrupal/localgov_events/issues/97

**Comment by stephen-cox** [comment]
> Hi @stephen-cox could we provide documentation for existing installs to make these changes themselves. Are they straightforward to do?

It's a very simple change. For Guides, go to /admin/structure/types/manage/localgov_guides_overview/form-display

![image](https://user-images.githubusercontent.com/7189914/235941568-20e681ed-3d73-4c12-b1b6-8044245025ec.png)

Find the body field, click on the little gear wheel on the right to expand the options and check 'Always show the summary field'.

![image](https://user-images.githubusercontent.com/7189914/235941957-a52ad09f-0156-45ab-b254-506be1d7d798.png)

**Comment by willguv** [comment]
Great! Is there a risk of this being overwritten?

**Comment by stephen-cox** [comment]
> Great! Is there a risk of this being overwritten?

This is a configuration change and so the site config should be exported after making the change as it will be lost with the next config import if not.

**Comment by willguv** [comment]
Cheers Stephen. So let's make sure our instructions include this

**Comment by willguv** [comment]
### How to get this

- All new LGD sites will get this update automatically
- If you have an existing site, it's best to ask your dev to make these changes (they're simple and won't take long) and then export the config files so they're not overwritten in the future
- If you can see the "Structure" menu in the nav, you can make these changes yourself. Be careful as deleting any fields will delete all the content associated with it!

Go to Structure > Content types > the type you want to change (eg Guide page) > Manage form display

![Image](https://github.com/localgovdrupal/localgov/assets/708140/cbdfec65-58c1-4e43-806e-dae585a14a60)

Scroll down to the "Body" field and click on the cog icon

![Image](https://github.com/localgovdrupal/localgov/assets/708140/98702912-4a34-4608-8cd7-0583a1814673)

You'll see the settings for the Body field appear. Make sure "Always show the summary field" is switched on

![Image](https://github.com/localgovdrupal/localgov/assets/708140/24bef9ae-f058-4516-854c-d236163bc611)

Hit update and then save the page. Then visit an example of the content type you've changed to check it works OK

You should then export the config, and a dev is best placed to do this. If you'd like to learn more, read [Drupal's config management section](https://www.drupal.org/docs/administering-a-drupal-site/configuration-management)

-----
How's this @stephen-cox ?
